### PR TITLE
compositor: Disable `IN_FENCE_FD` on Nvidia

### PR DESCRIPTION
**Pull request by ids1024**
Atomic commits fail if this is set on the Nvidia driver, it seems: https://github.com/NVIDIA/open-gpu-kernel-modules/issues/622

Kwin also has a check for this https://invent.kde.org/plasma/kwin/-/merge_requests/4770.

This fixes Anvil and cosmic-comp on the Nvidia 555 beta driver. (I'm not sure why it only started to be a problem; `EGL_ANDROID_native_fence_sync` is present in the 550 driver, etc.)

### CODE REVIEW

File: src/backend/drm/compositor/mod.rs [hunk 1]
```diff
@@ -1640,6 +1640,22 @@ where
...
+            }))
+        })?;
+        // `IN_FENCE_FD` makes commit fail on Nvidia driver
+        // https://github.com/NVIDIA/open-gpu-kernel-modules/issues/622
+        let is_nvidia = driver.name().to_string_lossy().to_lowercase().contains("nvidia")
+            || driver
+                .description()
+                .to_string_lossy()
+                .to_lowercase()
+                .contains("nvidia");
```
Comment (Drakulix) [inline src/backend/drm/compositor/mod.rs:1657]: Does the open-driver also set `nvidia` or is it possibly `nvidia-open` and we need to filter for that as well?

**Review by Drakulix — COMMENTED** [review]
> I'm not sure why it only started to be a problem; EGL_ANDROID_native_fence_sync is present in the 550 driver, etc.

Possibly because of the DrmCapability SyncObj now being announced?

**Comment by ids1024** [comment]
Ah yeah, it's probably the `DriverCapability::SyncObj`.

File: src/backend/drm/compositor/mod.rs (hunk 1 above)
Comment (ids1024) [reply src/backend/drm/compositor/mod.rs:1657]: https://github.com/NVIDIA/open-gpu-kernel-modules/blob/083cd9cf17ab95cd6f9fb50a5349c21eaa2f7d4b/kernel-open/nvidia-drm/nvidia-drm-drv.c#L1666-L1669

Looks like NVIDIA's open kernel modules just advertise the driver as `"nvidia-drm"`, `"NVIDIA DRM driver"`, `"20160202"`. Like the closed source kernel driver does.

**Review by Drakulix — APPROVED** [review]

**Pull request merged**